### PR TITLE
Add an unsafe_formatting code path

### PR DESCRIPTION
**Pull request by mrallen1**
This branch adds a new `_unsafe` path for log messages with a reasonable bounded size when its not necessary for lager to ensure that log messages don't kill an application because they're so huge.

### CODE REVIEW

**Comment by mrallen1** [comment]
Inspired by #273

**Comment by mrallen1** [comment]
@JeetKunDoug @jvoegele After our discussion last week would you please review this PR so it can be merged into 3.x?

**Comment by mrallen1** [comment]
cc @macintux

File: src/lager.erl [hunk 1]
```diff
@@ -486,6 +504,23 @@ safe_format(Fmt, Args, Limit, Options) ->
...
     safe_format(Fmt, Args, Limit, [{chomp, true}]).
 
+%% @private Print the format string `Fmt' with `Args' without a size limit.
+%% This is unsafe because the output of this function is unbounded.
+%%
+%% Log messages with unbounded size will kill your application dead as
+%% OTP mechanisms stuggle to cope with them.  So this function is
+%% intended <b>only</b> for messages which have a reasonable bounded
+%% size before they're formatted.
+%%
```
Comment (JeetKunDoug) [inline src/lager.erl]: I wonder if this is really a strong enough warning, and if it shouldn't have a README change as well - would be more likely to be seen there. Is there a set of README changes planned to go with this?

**Comment by JeetKunDoug** [comment]
Other than the above note on warning/guidance for unsafe stuff (which I'd be happy to help write/review), this looks fine to me.

**Comment by jvoegele** [comment]
I agree with @JeetKunDoug's comment above, but other than that :+1:

**Comment by mrallen1** [comment]
Yeah, thanks for the reminder. I'll add a blurb to the README as a new commit here in a second.

**Comment by mrallen1** [comment]
OK. I will add some language to the README in a second as a new commit.

**Comment by mrallen1** [comment]
OK, pushed some text about `_unsafe`

**Comment by JeetKunDoug** [comment]
:+1: 2e81dc3

**Comment by mrallen1** [comment]
@borshop merge

**Pull request merged**